### ISSUE TITLE
Access denied for user 'user'@'206.***.***.**' on mysql 8

### ISSUE THREAD

**Issue by onemoreahmad** [hyn/multi-tenant#875]
Awesome package, thanks for sharing.

I'm trying to make this work with Digital Ocean managed MySQL 8 Databases. 
It works perfect everywhere else, local +forge PHP 7.4 and Mysql 8, but when I try to connect the application to [DO MySQL](https://www.digitalocean.com/products/managed-databases-mysql/), I got a strange error only when dealing with tenant websites (create/view) but the main system connection works just fine.

The error is saying 

> Access denied for user 'user'@'206.***.***.**'

where the IP 206.*** is the webserver IP, I have no idea why he is trying to connect using the webserver IP for the tenant website and not using what I put inside .env file which is DO database host & port info ?

Any idea on that would be appreciated. 
Thanks a lot.

_I'm deploying my server on Forge (PHP 7.4) & MySQL 8 from Digital Ocean_

**Comment by Miguel-Serejo** [comment]
>I have no idea why he is trying to connect using the webserver IP

The error states that the mysql server is refusing the connection _from_ your webserver's IP. Check your mysql user permissions.

**Comment by onemoreahmad** [comment]
@36864 Thanks for your reply. The issue was from the password generator, somehow it doesn't accept the new generated password, for example, if I changed the password in the file `src/Database/Connection.php` located in the package src files, to a hardcoded password it works! 

in line 320, if I changed this :
`$clone['password'] = $this->passwordGenerator->generate($website);`
to this :
`$clone['password'] = 'b4z2M6c1fp02v8mnb';`
it works!

What I ended up doing, change the connection to for each new tenant to the 'system' connection and disable create new mysql user for each tenant, but keep the tenant databases separated. This is how I like it anyway, one connection with one mysql user to access multi tenant databases.

I'll close the issue as it somehow is fixed for me, even if it's not a "real" fix I believe 😀
